src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -302,6 +302,28 @@ def minimize_corpus(self, target_path, arguments, input_dirs, output_dir,
     """
     raise NotImplementedError
 
+  def _get_smallest_crasher(self, workdir_path):
+    """Returns the path to the smallest crash in Centipede's |workdir_path|."""
+    if not os.path.isdir(workdir_path):
+      logs.log_error(f'Work directory does not exist: {workdir_path}')
+      return None
+    crashes_path = os.path.join(workdir_path, 'crashes')
+    if not os.path.isdir(crashes_path):
+      logs.log_error(f'Crashes directory does not exist: {crashes_path}')
+      return None
+
+    testcases = [
+        os.path.join(crashes_path, t)
+        for t in os.listdir(crashes_path)
+        if os.path.isfile(os.path.join(crashes_path, t))
+    ]
+    if not testcases:
+      logs.log_error(f'No crash testcases under: {crashes_path}')
+      return None
+
+    minimum_testcase = min(testcases, key=os.path.getsize)
+    return minimum_testcase
+
   def minimize_testcase(self, target_path, arguments, input_path, output_path,
```

Comment:
Hi @lszekeres, this PR supports testcase minimization for `Centipede`: The `input_path` takes the testcase to be minimized, and `output_path` points to the same testcase after minimization.
For example, the unittest minimizes `?f???u???z?` to `?fuz?` while maintaining its ability to reproduce the crash.

Does this match the feature you requested via email earlier?
Please let me know if I missed anything, thanks : )
I will push another PR for corpus minimization.

Reply:
LGTM, thanks!